DSharpPlus.Core/Rest/RatelimitBucket.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace DSharpPlus.Core.Rest
+{
+
+    // represents a single ratelimit bucket, against a single endpoint.
+    internal record RatelimitBucket
+    {
+        private const string UnlimitedRatelimitIdentifier = "unlimited";
+        private const int DefaultLimit = 1;
+        private const int DefaultRemaining = 0;
+
+        // represents the limit for this bucket - the maximum amount of requests this bucket will allow
+        public int Limit { get; internal set; } = DefaultLimit;
+
+#pragma warning disable CS0420
+        // stores how many requests remain for this ratelimit bucket.
+        public int Remaining
+        {
+            get => Volatile.Read(ref _remaining);
+            set => Volatile.Write(ref _remaining, value);
```

Comment:
Even if this is internal, is there a reason to have a public setter here? Or even a setter at all, really.